packages/wallet-adapter-react/src/WalletProvider.tsx

```diff
@@ -76,13 +81,62 @@ export const AptosWalletAdapterProvider: FC<AptosWalletProviderProps> = ({
   }, [walletCore]);
 
   useEffect(() => {
-    if (autoConnect) {
-      if (localStorage.getItem("AptosWalletName") && !connected) {
-        connect(localStorage.getItem("AptosWalletName") as string);
-      } else {
-        // if we dont use autoconnect set the connect is loading to false
-        setIsLoading(false);
-      }
+    // Only attempt to auto connect once per render and only if there are wallets
+    if (didAttemptAutoConnectRef.current || !walletCore?.wallets.length) {
+      return;
+    }
+    didAttemptAutoConnectRef.current = true;
```

Comment:
This is never set to false. I'm guessing this is by design but pointing out jic :)